Runtime/Client/BugSplatManager.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using UnityEngine;
+
+namespace BugSplatUnity.Runtime.Client
+{
+	public sealed class BugSplatManager : MonoBehaviour
+	{
+		[SerializeField]
+		[Tooltip("BugSplat configuration SerializedObject to instantiate BugSplat with.")]
+		private BugSplatOptions bugSplatOptions = null;
+
+		[SerializeField]
+		[Tooltip("Should the BugSplatManager be destroyed when a new scene is loaded?")]
+		private bool dontDestroyManagerOnSceneLoad = true;
+
+		[SerializeField]
+		[Tooltip("Register BugSplat to capture LogType.Exceptions on initialization.")]
+		private bool registerLogMessageRecieved;
+
+		public BugSplat BugSplat { get; private set; }
+
+		private void Awake()
+		{
+			ConfigureBugSplat();
+			if (dontDestroyManagerOnSceneLoad)
+			{
+				DontDestroyOnLoad(this);
+			}
+		}
+
+		/// <summary>
+		/// Function to instantiate BugSplat object based on BugSplatOptions.
+		/// </summary>
+		private void ConfigureBugSplat()
```

Comment:
let's get rid of this function and just move the implementation into the Awake function. this function isn't particularly large so having all the Awake stuff in one place feels a bit more readable